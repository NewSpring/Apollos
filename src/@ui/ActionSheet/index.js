export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
