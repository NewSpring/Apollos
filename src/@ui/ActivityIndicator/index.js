import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
