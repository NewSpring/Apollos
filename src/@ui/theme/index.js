export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
