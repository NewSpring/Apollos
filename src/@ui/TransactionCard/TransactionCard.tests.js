import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '@ui/theme';
import TransactionCard from './';

// eslint-disable-next-line
const data = {"scheduledTransactions":[{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c7afcc9a198531f600da6a3cee3d19957","start":"Wed Jan 17 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c526d99425e516edbab13c811336e7e0d","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c7c1a22154b5aec0a8244dde1c1f84cc7","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c01982aa09da7adee4f0af2860d7cba9d","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906cc2c8d1185251fca1270b33de8569bee5","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906cc383b0af126c72efb970f92d1a768bb9","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c363576d51642786da729200a9b0373ce","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c630bc12d2ed369f4a431d8b14065cff7","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906ccb8b6dbe1d00aac4bcef3d1423f0e7d1","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c04622e37843659c3ae5ed49b2ac4b3a2","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c48903e12fb87c34494b6a255235a30b3","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c80ac1816f05e4c308058173aa68e296f","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c589fd516bc033837b9215dd5c147a46f","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c1f443dc822f17d3f9e328d73eb57962e","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906ce945f9a63074b55d6f14d73d39c717c1","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c76a4158105382ca0558a978a01aa1196","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c77ab9de3855c16f97e7ec3926f189e52","start":"Tue Jan 16 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906ce9ca1d5ec3b92ea8f49c30d206b839fe","start":"Mon Jan 15 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c26273ea3350f468e464a8e9ebf892556","start":"Mon Jan 15 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"},{"id":"f0db7e3b0c95948e8fd3a8e9cfe3906c4133785a912ae2b1e38f79990b5191db","start":"Mon Jan 15 2018 16:00:00 GMT-0800 (PST)","details":[{"account":{"name":"General Fund","__typename":"FinancialAccount"},"amount":1,"__typename":"TransactionDetail"}],"transactions":[],"schedule":{"description":"One Time","__typename":"DefinedValue"},"__typename":"ScheduledTransaction"}],"savedPaymentMethods":[{"id":29239,"name":"cc good","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29246,"name":"Bank Account","payment":{"accountNumber":"1xxxx3123","paymentType":"ACH","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29247,"name":"cc2","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29248,"name":"cc3","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29249,"name":"cc3","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29250,"name":"cc30","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29251,"name":"cc31","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29252,"name":"cc expiring","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29253,"name":"cc expiring 2","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"},{"id":29254,"name":"cc expiring 3","payment":{"accountNumber":"411111******1111","paymentType":"Visa","__typename":"PaymentDetail"},"__typename":"SavedPayment"}],"activityItems":[{"id":"01428de2c6ffc1850a90eaa51e840c7551b51ad12d91391f11a44c75022504d8","date":"2018-01-18T15:31:55.427Z","summary":"Reference Number: 3953283923","status":null,"statusMessage":null,"schedule":null,"details":[{"amount":1,"account":{"name":"General Fund","__typename":"FinancialAccount"},"__typename":"TransactionDetail"}],"__typename":"Transaction"},{"id":"01428de2c6ffc1850a90eaa51e840c75089c7a406d3a2f1ae70c0333cb815017","date":"2018-01-18T15:30:55.427Z","summary":"Reference Number: 3953281911","status":null,"statusMessage":null,"schedule":null,"details":[{"amount":1,"account":{"name":"General Fund","__typename":"FinancialAccount"},"__typename":"TransactionDetail"}],"__typename":"Transaction"},{"id":"01428de2c6ffc1850a90eaa51e840c75032f5d31e11720c2c10debdd45044bf3","date":"Wed Jan 17 2018 10:11:07 GMT-0800 (PST)","summary":"Reference Number: 3952129674","status":null,"statusMessage":null,"schedule":null,"details":[{"amount":1,"account":{"name":"General Fund","__typename":"FinancialAccount"},"__typename":"TransactionDetail"}],"__typename":"Transaction"},{"name":"cc expiring","expirationYear":"18","expirationMonth":"02","__typename":"SavedPayment"},{"name":"cc expiring 3","expirationYear":"18","expirationMonth":"02","__typename":"SavedPayment"}]};

describe('the TransactionCard component', () => {
  it('should render Success', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <TransactionCard
          date="02/02/2020"
          iconSize={20}
          dateFormat="lll"
          status="Success"
          details={data.scheduledTransactions[0].details}
          isScheduled
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render Failure', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <TransactionCard
          date="02/02/2020"
          iconSize={20}
          dateFormat="lll"
          status="Failed"
          details={data.scheduledTransactions[0].details}
          error="Unable to process transaction"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render scheduled payments', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <TransactionCard
          date="02/02/2020"
          iconSize={20}
          dateFormat="lll"
          status="Pending"
          details={data.scheduledTransactions[0].details}
          isScheduled
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});

