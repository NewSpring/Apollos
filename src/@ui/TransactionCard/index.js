import React from 'react';
import {
  View,
  TouchableWithoutFeedback,
} from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import moment from 'moment';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card from '@ui/Card';
import PaddedView from '@ui/PaddedView';
import { H5, H6 } from '@ui/typography';
import Icon from '@ui/Icon';
import { withTheme } from '@ui/theme';
import Spacer from '@ui/Spacer';
import StatusMessage from './StatusMessage';

const enhance = compose(
  setPropTypes({
    isLoading: PropTypes.bool,
    date: PropTypes.string,
    iconSize: PropTypes.number,
    dateFormat: PropTypes.string,
    status: PropTypes.string,
    details: PropTypes.arrayOf(PropTypes.shape({})),
    error: PropTypes.string,
    isScheduled: PropTypes.bool,
  }),
  defaultProps({
    isLoading: false,
    date: null,
    iconSize: null,
    dateFormat: 'MMM D YYYY',
    status: null,
    details: [],
    error: null,
    isScheduled: false,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
    isErrorColor: theme.colors.alert,
  })),
  pure,
);

const StyledH6 = styled(({ theme }) => ({
  color: theme.colors.text.link,
}))(H6);

const Row = styled({
  flexDirection: 'row',
  alignItems: 'center',
})(View);

const TransactionCard = enhance(({
  isLoading,
  iconSize,
  date,
  dateFormat,
  onPress,
  status,
  details,
  isScheduled,
  error,
  isErrorColor,
  ...otherProps
}) => {
  const isOk = status === null || status === 'Success' || status === 'Complete' || status === 'Pending';
  return (
    <Card isLoading={isLoading} {...otherProps}>
      <PaddedView>
        <Row>
          {isOk ? <Icon name="circle-outline-check-mark" size={iconSize} /> : <Icon name="circle-outline-x-mark" size={iconSize} fill={isErrorColor} />}
          <Spacer byWidth />
          <H5>{moment(date).utc().format(dateFormat)}</H5>
        </Row>
        <Spacer />
        <StatusMessage
          status={status}
          details={details}
          isScheduled={isScheduled}
          error={error}
        />
        <Spacer />
        <TouchableWithoutFeedback
          onPress={onPress}
        >
          <Row>
            <StyledH6>{'View Contribution'}</StyledH6>
            <Icon name="arrow-next" size={iconSize} />
          </Row>
        </TouchableWithoutFeedback>
      </PaddedView>
    </Card>
  );
});

export default TransactionCard;
