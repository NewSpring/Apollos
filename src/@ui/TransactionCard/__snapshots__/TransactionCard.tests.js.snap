// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TransactionCard component should render Failure 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Svg
            height={20}
            viewBox="0 0 60 61"
            width={20}
          >
            <Path
              d="M30,54.33 L30,54.33 C43.254834,54.33 54,43.584834 54,30.33 C54,17.075166 43.254834,6.33 30,6.33 C16.745166,6.33 6,17.075166 6,30.33 C6,43.584834 16.745166,54.33 30,54.33 L30,54.33 Z M30,60.33 L30,60.33 C13.4314575,60.33 0,46.8985425 0,30.33 C0,13.7614575 13.4314575,0.33 30,0.33 C46.5685425,0.33 60,13.7614575 60,30.33 C60,46.8985425 46.5685425,60.33 30,60.33 L30,60.33 Z"
              fill="#c64f55"
            />
            <Path
              d="M28.784,33.438 L28.784,33.438 L27.9222343,33.438 L27.7949654,32.5856839 L26.7389654,25.5136839 L26.728,25.366 L26.728,16.502 L26.728,15.502 L27.728,15.502 L32.272,15.502 L33.272,15.502 L33.272,16.502 L33.272,25.366 L33.2610346,25.5136839 L32.2050346,32.5856839 L32.0777657,33.438 L31.216,33.438 L28.784,33.438 Z M26.504,41.662 L26.504,41.662 C26.504,39.6282997 27.9530061,38.166 30,38.166 C32.0469939,38.166 33.496,39.6282997 33.496,41.662 C33.496,43.6957003 32.0469939,45.158 30,45.158 C27.9530061,45.158 26.504,43.6957003 26.504,41.662 L26.504,41.662 Z"
              fill="#c64f55"
            />
          </Svg>
          <View
            byWidth={true}
            style={
              Object {
                "height": null,
                "width": 10,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            Jan 17, 2018 12:00 AM
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#505050",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          Your 
          
          contribution to
          <Text
            accessible={true}
            allowFontScaling={true}
            bold={true}
            ellipsizeMode="tail"
            italic={false}
            style={
              Object {
                "color": "#505050",
                "fontFamily": "DroidSerif-Bold",
                "fontSize": 20.02,
                "lineHeight": 33.11,
                "paddingBottom": 2.25,
                "paddingTop": 1.8,
              }
            }
          >
             General Fund 
          </Text>
           
          was unsuccessful.
           Unfortunately, Unable to process transaction.
        </Text>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#6bac43",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            View Contribution
          </Text>
          <Svg
            height={20}
            viewBox="0 0 24 24"
            width={20}
          >
            <Path
              d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
              fill="#6bac43"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the TransactionCard component should render Success 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Svg
            height={20}
            viewBox="0 0 60 61"
            width={20}
          >
            <Polygon
              fill="#6bac43"
              points="25.9570657 39.7210789 26.6666667 40.4357125 27.3762676 39.7210789 40.709601 26.293183 41.4092427 25.5885792 40.709601 24.8839754 38.0429343 22.1983962 37.3333333 21.4837626 36.6237324 22.1983962 25.9570657 32.940713 27.3762676 32.940713 23.3762676 28.9123442 22.6666667 28.1977106 21.9570657 28.9123442 19.290399 31.5979234 18.5907573 32.3025272 19.290399 33.007131"
            />
            <Path
              d="M30,54.903 L30,54.903 C43.254834,54.903 54,44.157834 54,30.903 C54,17.648166 43.254834,6.903 30,6.903 C16.745166,6.903 6,17.648166 6,30.903 C6,44.157834 16.745166,54.903 30,54.903 L30,54.903 Z M30,60.903 L30,60.903 C13.4314575,60.903 0,47.4715425 0,30.903 C0,14.3344575 13.4314575,0.903 30,0.903 C46.5685425,0.903 60,14.3344575 60,30.903 C60,47.4715425 46.5685425,60.903 30,60.903 L30,60.903 Z"
              fill="#6bac43"
            />
          </Svg>
          <View
            byWidth={true}
            style={
              Object {
                "height": null,
                "width": 10,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            Jan 17, 2018 12:00 AM
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#505050",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          Your 
          scheduled 
          contribution of 
          <Text
            accessible={true}
            allowFontScaling={true}
            bold={false}
            ellipsizeMode="tail"
            italic={false}
            style={
              Object {
                "color": "#505050",
                "fontFamily": "DroidSerif-Regular",
                "fontSize": 20.02,
                "lineHeight": 33.11,
                "paddingBottom": 2.25,
                "paddingTop": 1.8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              bold={true}
              ellipsizeMode="tail"
              italic={false}
              style={
                Object {
                  "color": "#505050",
                  "fontFamily": "DroidSerif-Bold",
                  "fontSize": 20.02,
                  "lineHeight": 33.11,
                  "paddingBottom": 2.25,
                  "paddingTop": 1.8,
                }
              }
            >
              1.00
            </Text>
             to 
            <Text
              accessible={true}
              allowFontScaling={true}
              bold={true}
              ellipsizeMode="tail"
              italic={false}
              style={
                Object {
                  "color": "#505050",
                  "fontFamily": "DroidSerif-Bold",
                  "fontSize": 20.02,
                  "lineHeight": 33.11,
                  "paddingBottom": 2.25,
                  "paddingTop": 1.8,
                }
              }
            >
              General Fund
            </Text>
          </Text>
           was successful.
        </Text>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#6bac43",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            View Contribution
          </Text>
          <Svg
            height={20}
            viewBox="0 0 24 24"
            width={20}
          >
            <Path
              d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
              fill="#6bac43"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the TransactionCard component should render scheduled payments 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Svg
            height={20}
            viewBox="0 0 60 61"
            width={20}
          >
            <Polygon
              fill="#6bac43"
              points="25.9570657 39.7210789 26.6666667 40.4357125 27.3762676 39.7210789 40.709601 26.293183 41.4092427 25.5885792 40.709601 24.8839754 38.0429343 22.1983962 37.3333333 21.4837626 36.6237324 22.1983962 25.9570657 32.940713 27.3762676 32.940713 23.3762676 28.9123442 22.6666667 28.1977106 21.9570657 28.9123442 19.290399 31.5979234 18.5907573 32.3025272 19.290399 33.007131"
            />
            <Path
              d="M30,54.903 L30,54.903 C43.254834,54.903 54,44.157834 54,30.903 C54,17.648166 43.254834,6.903 30,6.903 C16.745166,6.903 6,17.648166 6,30.903 C6,44.157834 16.745166,54.903 30,54.903 L30,54.903 Z M30,60.903 L30,60.903 C13.4314575,60.903 0,47.4715425 0,30.903 C0,14.3344575 13.4314575,0.903 30,0.903 C46.5685425,0.903 60,14.3344575 60,30.903 C60,47.4715425 46.5685425,60.903 30,60.903 L30,60.903 Z"
              fill="#6bac43"
            />
          </Svg>
          <View
            byWidth={true}
            style={
              Object {
                "height": null,
                "width": 10,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            Jan 17, 2018 12:00 AM
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          bold={false}
          ellipsizeMode="tail"
          italic={false}
          style={
            Object {
              "color": "#505050",
              "fontFamily": "DroidSerif-Regular",
              "fontSize": 20.02,
              "lineHeight": 33.11,
              "paddingBottom": 2.25,
              "paddingTop": 1.8,
            }
          }
        >
          Your 
          scheduled 
          contribution to
          <Text
            accessible={true}
            allowFontScaling={true}
            bold={true}
            ellipsizeMode="tail"
            italic={false}
            style={
              Object {
                "color": "#505050",
                "fontFamily": "DroidSerif-Bold",
                "fontSize": 20.02,
                "lineHeight": 33.11,
                "paddingBottom": 2.25,
                "paddingTop": 1.8,
              }
            }
          >
             General Fund 
          </Text>
           
          is 
          <Text
            accessible={true}
            allowFontScaling={true}
            bold={false}
            ellipsizeMode="tail"
            italic={false}
            style={
              Object {
                "color": "#505050",
                "fontFamily": "DroidSerif-Regular",
                "fontSize": 20.02,
                "lineHeight": 33.11,
                "paddingBottom": 2.25,
                "paddingTop": 1.8,
              }
            }
          >
            pending
          </Text>
          .
        </Text>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#6bac43",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            View Contribution
          </Text>
          <Svg
            height={20}
            viewBox="0 0 24 24"
            width={20}
          >
            <Path
              d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
              fill="#6bac43"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</View>
`;
